moshi/moshi/run_tts.py

```diff
@@ -0,0 +1,210 @@
+        effective_duration = 0.
+        for idx, request in enumerate(batch):
+            end_step = result.end_steps[idx]
+            if end_step is None:
+                print(f"Warning: end step is None, generation failed for {request.id}")
+                wav_length = wavs.shape[-1]
+            else:
+                wav_length = int((mimi.sample_rate * (end_step + tts_model.final_padding) / mimi.frame_rate))
+            effective_duration += wav_length / mimi.sample_rate
+            wav = wavs[idx, :, :wav_length]
+            start_step = 0
+            if prefixes is not None:
+                start_step = prefixes[idx].shape[-1]
+                start = int(mimi.sample_rate * start_step / mimi.frame_rate)
+                wav = wav[:, start:]
+            filename = args.out_folder / f"{request.id}.wav"
+            debug_tensors = {
+                'frames': frames[idx].short(),
+            }
+            sphn.write_wav(filename, wav.clamp(-1, 1).cpu().numpy(), mimi.sample_rate)
+            save_file(debug_tensors, filename.with_suffix('.safetensors'))
+            debug_info = {
+                'hf_repo': args.hf_repo,
+                'voice_repo': args.voice_repo,
+                'model_id': checkpoint_info.model_id,
+                'cfg_coef': tts_model.cfg_coef,
+                'temp': tts_model.temp,
+                'max_padding': tts_model.machine.max_padding,
+                'initial_padding': tts_model.machine.initial_padding,
+                'final_padding': tts_model.final_padding,
+                'padding_between': args.padding_between,
+                'padding_bonus': tts_model.padding_bonus,
+                'transcript': result.all_transcripts[idx],
+                'consumption_times': result.all_consumption_times[idx],
+                'turns': request.turns,
+                'voices': request.voices,
+                'logged_text_tokens': result.logged_text_tokens[idx],
+                'end_step': end_step,
+                'start_step': start_step,
+            }
+            with open(filename.with_suffix('.json'), 'w') as f:
+                json.dump(debug_info, f)
+            print("Saved", filename)
+        time_taken = time.time() - begin
+        total_speed = total_duration / time_taken
+        effective_speed = effective_duration / time_taken
+        # Total speed is the speed we get assuming all the item in the batch are the same length.
+        # However, some items might have finished earlier, in which case the computation for those
+        # was wasted. Effective speed accounts for that, and gives the speed up accounting for
+        # the actual amount of usable audio generated.
+        print(f"[TOT] Batch of size {len(batch)} took {time_taken:.2f}s, "
+              f"total speed {total_speed:.2f}x, "
+              f"effective speed {effective_speed:.2f}x")
+        batch.clear()
+
+    batch: list[TTSRequest] = []
+    with args.jsonl.open('r') as f:
```

Comment:
It would probably be nice to include a sample `jsonl` in the repo so as to have an easy way to try things out and know the expected format.